### ISSUE TITLE
Error on connect Firebird 3

### ISSUE THREAD

**Issue by Level0r0s** [dbeaver/dbeaver#2424]
Driver jaybird-jdk17-3.0.2.jar  no compatible with firebird 3

Error:  Incompatible wire encryption levels requested on client and server  [SQLState:28000, ISC erros code:335545064]

help

**Comment by serge-rider** [comment]
You need to update driver version to 3.x:
Open driver edit dialog->Download/Update->Click on `Version` column. Select 3.x version in dropdown.

![image](https://user-images.githubusercontent.com/6398845/32138060-212417dc-bc35-11e7-8308-054255c82571.png)

**Comment by hdjong** [comment]
I have installed 3.0.1 driver. It still does not work.Get the same error description. I use Windows10. Please help. Tried the other drivers also.

**Comment by jahudka** [comment]
Hi, I'm experiencing this issue as well running DBeaver 4.3.1 on macOS High Sierra with all of the 3.0.x versions of the Firebird driver up to 3.0.3. Versions 3.0.0 and 3.0.1 first complain about connection charset property, but if I specify that explicitly in the driver settings then the same behaviour as with 3.0.2 and 3.0.3 occurs, which is that the connection fails with the above mentioned error. I tried specifying the `wireCrypt` connection property in the driver settings with various values as per the [Jaybird documentation](https://github.com/FirebirdSQL/jaybird/blob/master/src/documentation/release_notes.md#wire-encryption-support) to no avail.

**Comment by bsgmbh** [comment]
Hi, I just wanted to pitch in and say that I am seeing the exact same behavior as the previous poster, nothing to add to the description, just exactly the same sequence with 3.0.0 complaining about charset and once that is set with lc_ctype=UTF8, then the second WireCrypt related error kicks in. Same with 32bit and 64bit versions of dBeaver.

**Comment by jahudka** [comment]
I'd add that the server I'm attempting to connect to is running the latest Firebird TCP/IP server version LI-V3.0.1.32609 from the Debian Stretch repos.

**Comment by bsgmbh** [comment]
Mine is a Win32 V3.0.1.32609 that came preinstalled with my ERP system. Also, to qualify my statement above, both Firebird 3.0.0 and 3.0.1 drivers show the lc_ctype behavior and then kick into WireCrypt error mode and 3.0.2 and 3.0.3 do not show the lc_ctype behavior but instantly show the WireCrypt error.

**Comment by serge-rider** [comment]
Out test FB server is 3.0.2.32703.

We have this in firebird.conf:
```
WireCrypt = Disabled
```
This removes errors about incompatible wireCrypt. 
I didn't find any combination of wireCrypt values in client/server configurations which work (besides disabled wireCrypt).

Also drivers 3.0.0 and 3.0.1 complain about lc_type, newer versions just print warning in log: 
```
WARNING: No connection character set specified (property lc_ctype, encoding, charSet or localEncoding), defaulting to character set NONE`).
```
I'll fix default driver config and set encoding to UTF8 by default.

**Comment by ArloL** [comment]
The default setting for WireCrypt is incompatible with jaybird: https://github.com/FirebirdSQL/jaybird/wiki/Jaybird-and-Firebird-3#jaybird-3

**Comment by bubibubi** [comment]
@mrotteveel where should I put WireCrypt = Enabled? Adding it in driver properties does not work

**Comment by stayen** [comment]
Confirming the bug presence when trying to connect version 5.3.2 (CE) to Firebird 3 with default settings.

Only by setting

    WireCrypt = disabled

I can use Dbeaver with that Firebird. If it could be possible to add control to Dbeaver to disable WireCrypt without altering server configuration, it could be nice to have (changing that setting at the server isn't always possible).

**Comment by mrotteveel** [comment]
@stayen Jaybird 3.0.4 or higher supports Firebird wire encryption (if you are using a `Srp` user, `Legacy_auth` users do not support wire encryption). 

Setting `WireCrypt = Enabled` is sufficient for earlier versions of Jaybird (or if you're using a `Legacy_auth` user. It has never been necessary to set it to `Disabled` and that will unnecessarily degrade security for clients that do support wire encryption.

Client applications cannot lower the security configuration of the server, so DBeaver cannot provide an option to disable this.

So in short: if you want to be able to use DBeaver with the default Firebird configuration, you'll need to use Jaybird 3.0.4 or higher (3.0.5 is the latest at time of writing), and you'll need to use a `Srp` user.
